File: C++/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& A, int x) {
        map<int,int> mp;
        vector<int> B;
        int n = A.size();
        for(int i=0;i<n;i++) {
            if(mp[x-A[i]] != 0) {
                B.push_back(i);
                B.push_back(mp[x-A[i]] - 1);
                return B;
            }
            mp[A[i]] = i+1;
        }
        return B;
    }
};

File: Java/1.java
class Solution {
    public int[] twoSum(int[] A, int x) {
        int[] ans = new int[2];
        Map<Integer,Integer> mp = new HashMap<>();
        for(int i=0;i<A.length;i++) {
            if(mp.getOrDefault(x - A[i],0) != 0) {
                ans[0] = i;
                ans[1] = mp.get(x - A[i]) - 1;
                return ans;
            }
            mp.put(A[i],i+1);
        }
        return ans;
    }
}
